### PR TITLE
Add CI leg to test strict indentation

### PR DESCRIPTION
**Pull request by auduchinok**
Adds a new CI leg to test the indentation/parser changes independently from other preview changes, as suggested in https://github.com/dotnet/fsharp/pull/15637#issuecomment-1643625484.

### CODE REVIEW

**Comment by auduchinok** [comment]
@vzarytovskii Could you check what's wrong with the new leg when you have time, please?
There's an error about publishing build artifacts, and I don't know what to do with it.

**Comment by vzarytovskii** [comment]
> @vzarytovskii Could you check what's wrong with the new leg when you have time, please? There's an error about publishing build artifacts, and I don't know what to do with it.

I'm actually not sure custom flag is passed there, I think build script might be treating it as its own flag, and not passed to msbuild?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -347,6 +347,33 @@ stages:
...
+            # and there is an alternate build definition that sets this to a queue that is always scouting the
+            # next preview of Visual Studio.
+            name: $(DncEngPublicBuildPool)
+            demands: ImageOverride -equals $(WindowsMachineQueueName)
+          timeoutInMinutes: 120
+          steps:
+          - checkout: self
+            clean: true
+
+          - script: eng\CIBuild.cmd -compressallmetadata -configuration Release /p:OtherFlags='$(OtherFlags) --strict-indentation+'
```
Comment (T-Gro) [inline azure-pipelines.yml:363]:
```suggestion
          - script: eng\CIBuild.cmd -compressallmetadata -configuration Release /p:AdditionalFscCmdFlags=--strict-indentation+
```

Comment (auduchinok) [reply azure-pipelines.yml]: @T-Gro Thanks!

**Comment by auduchinok** [comment]
This is ready.

Thanks for the help! 🙂

**Review by vzarytovskii — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -347,6 +347,33 @@ stages:
...
+
+          - script: eng\CIBuild.cmd -compressallmetadata -configuration Release /p:AdditionalFscCmdFlags=--strict-indentation+
+            displayName: Build
+
+          - task: PublishBuildArtifacts@1
+            displayName: Publish Build BinLog
+            condition: always()
+            continueOnError: true
+            inputs:
+                PathToPublish: '$(Build.SourcesDirectory)\artifacts\log/Release\Build.VisualFSharp.sln.binlog'
```
Comment (abonie) [inline azure-pipelines.yml:371]: This looks weird, is it supposed to have both forward and backward slashes?

**Review by abonie — APPROVED** [review]

File: azure-pipelines.yml (hunk 2 above)
Comment (vzarytovskii) [reply azure-pipelines.yml:371]: It is weird, but it is how it's been forever, so I didn't want to change it, since it was just working.

**Pull request merged**